clang/runtime/dpct-rt/include/dpct/image.hpp

```diff
@@ -840,6 +840,658 @@ static inline image_wrapper_base *create_image_wrapper(image_data data,
   return nullptr;
 }
 
+#ifdef SYCL_EXT_ONEAPI_BINDLESS_IMAGES
+
+using image_mem_p = sycl::ext::oneapi::experimental::image_mem *;
+
+namespace detail {
+struct sampled_image_handle_compare {
+  bool
+  operator()(sycl::ext::oneapi::experimental::sampled_image_handle L,
+             sycl::ext::oneapi::experimental::sampled_image_handle R) const {
+    return L.raw_handle < R.raw_handle;
+  }
+};
+
+inline std::pair<image_data, sampling_info> &get_img_info_map(
+    const sycl::ext::oneapi::experimental::sampled_image_handle handle) {
+  static std::map<sycl::ext::oneapi::experimental::sampled_image_handle,
+                  std::pair<image_data, sampling_info>,
+                  sampled_image_handle_compare>
+      img_info_map;
+  return img_info_map[handle];
+}
+
+inline image_mem_p &get_img_mem_map(
```

Comment:
for public interface let's use "sycl::ext::oneapi::experimental::image_mem *".

Reply:
fixed